lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1572,16 +1573,24 @@ LogicalResult FIRRTLLowering::visitDecl(WireOp op) {
   // Name attr is required on sv.wire but optional on firrtl.wire.
   auto nameAttr = op.nameAttr() ? op.nameAttr() : builder.getStringAttr("");
 
+  if (!AnnotationSet(op).hasDontTouch())
+    return setLoweringTo<sv::WireOp>(op, resultType, nameAttr);
+  auto moduleName =
+      cast<hw::HWModuleOp>(op->getParentOp()).getNameAttr().getValue().str();
```

Comment:
I think that HWModuleOp has an accessor that returns the string, e.g. `.name()` or `.getName()`?  Also, please leave "moduleName" as a StringRef instead of std::string.